### PR TITLE
update README.rst

### PR DESCRIPTION
**Pull request by newville**
Fixes README.rst to not reference Python 2.7, and provide other updates.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -29,6 +29,16 @@ platform specific code.
 .. contents:: **Contents**
 
 
+Development at GitHub.com
+-----------------------------
+
+Development of wxPython is done with git, using
+https://github.com/wxWidgets/Phoenix as the main devlopment site.
```
Comment (swt2c) [inline README.rst:36]: devlopment -> development

File: README.rst [hunk 2]
```diff
@@ -29,6 +29,16 @@ platform specific code.
...
+
+Currently (September, 2024), only 2 developers (@swt2c, @RobinD42) can merge
+pull requests, or assign others to review pull requests.
```
Comment (swt2c) [inline README.rst:39]: That last part isn't right.  Anyone can review pull requests without being assigned.

File: README.rst [hunk 3]
```diff
@@ -412,10 +412,8 @@ Phoenix.
 Help and Helping
 ----------------
 
-Most discussions about Phoenix happen on the wxPython-dev google group
-(a.k.a. the wxPython-dev mail list.) If you have questions or would like to
-get involved please subscribe to the group at
-https://groups.google.com/forum/#!forum/wxpython-dev and join in.
+Discussions about Phoenix developmet happen on this Github page.  Discussions
+about Python usage happen at https://discuss.wxpython.org/.
```
Comment (swt2c) [inline README.rst:416]: developmet -> development

I would maybe rephrase this section to something like "Please report bugs using GitHub issues.  For help using wxPython, please use the discussion forums at https://dicuss.wxpython.org/" or something similar.

Comment (newville) [reply README.rst]: OK, I updated that.  You might consider enabling Github Discussions too.

File: README.rst (hunk 1 above)
Comment (newville) [reply README.rst]: Thanks - fixed

File: README.rst (hunk 2 above)
Comment (newville) [reply README.rst:39]: I think it is right.  The statement is about who can assign other to review.   I think GH refers to this as "who can Manage and Triage" issues and pull requests.   These roles do not have to be an owner.    In many large projects, there are multiple people assigning reviewers, and all PRs need a review before merging.

Comment (swt2c) [reply README.rst:39]: OK so you mean specifically assigning reviewers, not reviewing itself.  OK.
